### PR TITLE
Update general UPP for GTG4

### LINKED ISSUE #763: GTG4 upgrade related
For RRFS, GTG4 is an upgrade version with the following features:
1. A new interface with UPP. To be compatible with GTG4, gtg_algo() in MDLFLD needs to be updated.
2. CMakeLists is updated for new GTG4 source code files
3. A new GTG field, convectively-induced turbulence (CIT), is introduced in GTG4.
4. tke is used for RRFS
5. In MISCLN.f, synchronizing the update of the interpolation to flight levels in general UPP for RRFS GTG related fields
6. To prepare separation of WAFS from GFS, new scripts are added
> scripts/exwafs_atmos_nceppost.sh
> ush/wafs_nceppost.sh
7. Add gtg_on option to the namelist 'NAMPGB'

### LINKED ISSUE #764: Clean out WAFS fields on ICAO standard atmospheric levels
Remove WAFS fields on ICAO standard atmospheric levels from post_avblflds.xml
And remove MDL2STD_P.f which does interpolation on ICAO standard atmospheric levels

### PR DESCRIPTION
**Pull request by YaliMao-NOAA**
1. Change gtg_algo() interface in MDLFLD.f
2. Add CIT(:,:,:) array for In-Cloud Turbulence, ALLOCATE_ALL.f VRBLS3D_mod.f DEALLOCATE.f
3. Modify CMakeLists.txt for updated/new GTG source code files.
4. Add stub code for the new GTG source	files in general UPP
5. Modify interpolation	in MISCLN.f for	RRFS GTG
6. Modify jobs/J_NCEPPOST to run WAFS downstream only
   Add scripts/exwafs_atmos_nceppost.sh
   Add ush/wafs_nceppost.sh
7. Add GEFS WAFS control files (.xml .txt)
8. (As an update later on) Add gtg_on option
9. Remove WAFS fields on ICAO standard atmospheric levels from post_avblflds.xml
10. Remove MDL2STD_P.f which does interpolation on ICAO standard atmospheric levels which is not ICAO required.

### CODE REVIEW

**Comment by WenMeng-NOAA** [comment]
@YaliMao-NOAA Do you have an UPP issue linked this PR? If not, please open a new one.

**Comment by YaliMao-NOAA** [comment]
@WenMeng-NOAA I opened an issue https://github.com/NOAA-EMC/UPP/issues/763

**Comment by WenMeng-NOAA** [comment]
@YaliMao-NOAA Please add a change log with your name in the fortran code files where if change log sessions are  available.

**Comment by YaliMao-NOAA** [comment]
> @YaliMao-NOAA Please add a change log with your name in the fortran code files where if change log sessions are available.

You mean the code change history?

**Comment by WenMeng-NOAA** [comment]
@YaliMao-NOAA See the example at
https://github.com/NOAA-EMC/UPP/blob/develop/sorc/ncep_post.fd/INITPOST_NETCDF.f#L42

**Comment by YaliMao-NOAA** [comment]
@WenMeng-NOAA I added change logs to the Fortran source code files. 
Also I added gtg_on option to the namelist

**Comment by WenMeng-NOAA** [comment]
@YaliMao-NOAA My UPP with gtg code build process failed as:
```
CMake Error at sorc/ncep_post.fd/CMakeLists.txt:194 (add_library):
  Cannot find source file:

    post_gtg.fd/gtg_cdf_mod.F90

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .h .hh .h++
  .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .ispc


CMake Error at sorc/ncep_post.fd/CMakeLists.txt:194 (add_library):
  No SOURCES given to target: upp


CMake Generate step failed.  Build files cannot be regenerated correctly
```
There is no file "gtg_cdf_mod.F90"  from gtg code checkout with the revision d0ad324.

**Comment by YaliMao-NOAA** [comment]
Sorry, I should have update UPP_GTG develop branch. I am requesting a viewer to approve a merge pull request of GTG code.

File: sorc/ncep_post.fd/WRFPOST.f [hunk 1]
```diff
@@ -36,6 +36,8 @@
 !> 2023-04-04 | Li(Kate Zhang)  |Add namelist optoin for CCPP-Chem (UFS-Chem) 
 !         and 2D diag. output (d2d_chem) for GEFS-Aerosols and CCPP-Chem model.
 !> 2023-05-20 | Rahul Mahajan             | Bug fix for fileNameFlat as namelist configurable
+!> 2023-05-20 | Rahul Mahajan             | Bug fix for fileNameFlat as namelist configurable
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/WRFPOST.f]: @YaliMao-NOAA Please remove this duplicate line.

**Comment by WenMeng-NOAA** [comment]
@YaliMao-NOAA Please verify my UPP standalone test results at /u/wen.meng/ptmp/gfs.2019083000 on Cactus and let me know if you saw issues.

**Comment by WenMeng-NOAA** [comment]
@YaliMao-NOAA The UPP RT failed in all test case as:
```
+ mpiexec -l -n 96 -ppn 48 /u/wen.meng/ptmp/yali/UPP/exec/upp.x
nid003523.cactus.wcoss2.ncep.noaa.gov 2: forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source          
upp.x              0000000000F4F36A  Unknown               Unknown  Unknown
libpthread-2.31.s  0000150A256408C0  Unknown               Unknown  Unknown
upp.x              0000000000861FAA  mdl2p_                    373  MDL2P.f
libiomp5.so        0000150A26A6D3F3  __kmp_invoke_micr     Unknown  Unknown
libiomp5.so        0000150A269F1937  __kmp_fork_call       Unknown  Unknown
libiomp5.so        0000150A269B5533  __kmpc_fork_call      Unknown  Unknown
upp.x              000000000081986E  mdl2p_                    322  MDL2P.f

```
I guess you need to add new option "gtg_on"  in these kind of routine for skipping gtg related computations.

**Comment by YaliMao-NOAA** [comment]
The test result looks good to me /u/wen.meng/ptmp/gfs.2019083000

As to the failure of SIGSEGV on your RT, you are right, gtg_on needs to be turned on by 'gtg_on=.true.' Do you think I should include 'gtg_on' option to avoid this kind of errors in mdl2p?

**Comment by WenMeng-NOAA** [comment]
> The test result looks good to me /u/wen.meng/ptmp/gfs.2019083000
> 
> As to the failure of SIGSEGV on your RT, you are right, gtg_on needs to be turned on by 'gtg_on=.true.' Do you think I should include 'gtg_on' option to avoid this kind of errors in mdl2p?

@YaliMao-NOAA Yes, you might also check other gtg related routines.

**Comment by YaliMao-NOAA** [comment]
I modified the code, and added a new issue related to this PR. https://github.com/NOAA-EMC/UPP/issues/764

**Comment by WenMeng-NOAA** [comment]
> I modified the code, and added a new issue related to this PR. #764

@YaliMao-NOAA Can you note your new changes at the beginning session of this PR, such as
1. Change gtg_algo() interface in MDLFLD.f
2. Add CIT(:,:,:) array for In-Cloud Turbulence, ALLOCATE_ALL.f VRBLS3D_mod.f DEALLOCATE.f
...

**Comment by WenMeng-NOAA** [comment]
@YaliMao-NOAA You need to tune your branch further. Here are the latest errors from the UPP RT for nmmb:
```
upp.x              00000000008626A8  mdl2p_                    387  MDL2P.f
libiomp5.so        000014EA165423F3  __kmp_invoke_micr     Unknown  Unknown
libiomp5.so        000014EA164C6937  __kmp_fork_call       Unknown  Unknown
libiomp5.so        000014EA1648A533  __kmpc_fork_call      Unknown  Unknown
upp.x              0000000000819CBC  mdl2p_                    335  MDL2P.f
upp.x              0000000000612D37  process_                   78  PROCESS.f
upp.x              000000000056D7FC  MAIN__                    701  WRFPOST.f
```
The whole runtime log file can be found /u/wen.meng/noscrub/ncep_post/post_regression_test_new/jobs/out.nmmb.test.
You might run the UPP RT by yourself for debugging. The UPP RT procedure can be found at 
https://docs.google.com/document/d/1gtk1DBxEa9GsaWbxrxtB4Ot7U_lNIHjLXaU2F6FyD7M/edit

**Comment by YaliMao-NOAA** [comment]
@WenMeng-NOAA It turns out gtg_interpolation was not initialized in MDL2P.f, I fixed it. Thank you for recommendation to do the regression test by myself. All regression tests passed. FYI, for rtma test, should it be 'cd $workdir' instead of '$homedir/jobs'?

File: sorc/ncep_post.fd/PROCESS.f [hunk 2]
```diff
@@ -115,11 +116,11 @@ SUBROUTINE PROCESS(kth,kpv,th,pv,iostatusD3D)
 !     COMPUTE/POST TROPOPAUSE DATA, FD LEVEL FIELDS,
 !     FREEZING LEVEL HEIGHT AND RH, BOUNDARY LAYER FIELDS,
 !     AND LFM-NGM LOOK-ALIKE FIELDS.
-      btim = mpi_wtime()
-      CALL MDL2STD_P
-      if(me==0) write(*,*) "PROCESS MDL2STD_P done"
-      MDL2STD_tim = MDL2STD_tim +(mpi_wtime() - btim)
-!
+!      btim = mpi_wtime()
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/PROCESS.f]: @YaliMao-NOAA Could you remove these comment lines since the routine MDL2STD_P was deleted from UPP code in your PR?

**Comment by WenMeng-NOAA** [comment]
@YaliMao-NOAA With new version of gtg4 implemented, do you think the gtg files gtg.config.gfs  gtg_imprintings.txt under UPP/parm should be removed?

**Comment by WenMeng-NOAA** [comment]
The UPP RT is completed on WCOSS2 with no changed results.

**Comment by WenMeng-NOAA** [comment]
@FernandoAndrade-NOAA You may start the UPP RT on Hera and Orion. There will no baseline update with this PR.

**Comment by FernandoAndrade-NOAA** [comment]
Apologies for the delay, Orion UPP RTs have completed successfully with no changes to results. Hera should finish momentarily.

**Comment by FernandoAndrade-NOAA** [comment]
Hera UPP RTs have completed as well with no changes to results.

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
@gspetro-NOAA Please be aware of the updates in parm/post_avblfld.xml in this PR for your new release of the UPP user guide later.

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
This PR is ready for merging.

**Pull request merged**